src/EnvelopeCollection.php

```diff
@@ -13,33 +13,28 @@
 
 use Symfony\Component\Messenger\Envelope;
 use Zenstruck\Assert;
+use Zenstruck\Messenger\Test\Bus\TestBus;
 use Zenstruck\Messenger\Test\Transport\TestTransport;
 
 /**
  * @author Kevin Bond <kevinbond@gmail.com>
  *
  * @implements \IteratorAggregate<TestEnvelope>
  */
-final class EnvelopeCollection implements \IteratorAggregate, \Countable
+abstract class EnvelopeCollection implements \IteratorAggregate, \Countable
 {
-    private TestTransport $transport;
-
     /** @var Envelope[] */
     private array $envelopes;
 
     /**
      * @internal
      */
-    public function __construct(TestTransport $transport, Envelope ...$envelopes)
+    public function __construct(Envelope ...$envelopes)
     {
-        $this->transport = $transport;
         $this->envelopes = $envelopes;
     }
 
-    public function back(): TestTransport
-    {
-        return $this->transport;
-    }
+    abstract public function back(): TestBus|TestTransport;
```

Comment:
Let's just remove this abstract method.

Reply:
https://github.com/zenstruck/messenger-test/pull/54#discussion_r1094462734

Reply:
I also suggested changes to the TestTransport return types below that should solve this.